tests/StreamTest.php

```diff
@@ -68,6 +68,24 @@ public function testConvertsToString()
         $stream->close();
     }
 
+    public function testConvertsToStringNonSeekableStream()
+    {
+        $handle = popen('echo foo', 'r');
+        $stream = new Stream($handle);
+        $this->assertFalse($stream->isSeekable());
+        $this->assertSame('foo', trim((string) $stream));
+    }
+
+    public function testConvertsToStringNonSeekablePartiallyReadStream()
+    {
+        $handle = popen('echo bar', 'r');
+        $stream = new Stream($handle);
+        $firstLetter = $stream->read(1);
+        $this->assertFalse($stream->isSeekable());
+        $this->assertSame('b', $firstLetter);
+        $this->assertSame('ar', trim((string) $stream));
```

Comment:
AFAIK the string cast is meant to return the whole stream from the beginning according to the RFC. This PR changes the behavior. Not sure if this is the right thing to do. But it seems indeed more reasonable to return the remaining contents than to return nothing.

Reply:
The main objective of this PR is to achieve consistency among the many available solutions. Please see the links below.

 - https://github.com/laminas/laminas-diactoros/blob/24f50cef61c0362bbf27e92cbc6fde9dbf219fb6/src/Stream.php#L73
 - https://github.com/Nyholm/psr7/blob/55ff6b76573f5b242554c9775792bd59fb52e11c/src/Stream.php#L101

You are right, cast a stream to a string should return the whole stream. But in case of non-seekable stream we can't to rewind it => I think it's a good practice to return the rest of the stream